openminers/base/blacklist.py

```diff
@@ -65,28 +62,33 @@ def default_blacklist(
     if forward_call.src_hotkey in self.config.miner.blacklist.blacklist:
         return True, "blacklisted hotkey"
 
-    # Check if the key is registered.
-    registered = False
-    if self.metagraph is not None:
-        registered = forward_call.src_hotkey in self.metagraph.hotkeys
-
-    # Check if we allow non-registered users.
-    if not registered:
-        return True, "hotkey not registered"
+    # Check registration if we do not allow non-registered users
+    if (
+        not self.config.miner.blacklist.allow_non_registered and
+        self.metagraph is not None and 
+        forward_call.src_hotkey not in self.metagraph.hotkeys
+    ):
+            return True, "hotkey not registered"
 
     # If the user is registered, it has a UID.
     uid = self.metagraph.hotkeys.index(forward_call.src_hotkey)
 
     # Check if the key has validator permit
     if (
-        self.metagraph.validator_permit[uid]
-        and self.config.miner.blacklist.force_validator_permit
+        self.config.miner.blacklist.force_validator_permit and
+        not self.metagraph.validator_permit[uid]
     ):
         return True, "validator permit required"
 
     if is_prompt_in_cache(self, forward_call):
         return True, "prompt already sent recently"
 
+    # request period
+    if forward_call.src_hotkey in self.request_timestamps:
+        period = time.time() - self.request_timestamps[forward_call.src_hotkey][0]
+        if period < self.config.miner.blacklist.min_request_period * 60:
```

Comment:
I see, so you are checking if there has been 50 requests within the last 30 mins?

Reply:
Perhaps just a small suggestion, its easier to understand if we only had a single param instead of 2 params that control the blacklist, ie. `--blacklist.requests_per_hour`

Reply:
Something like this?

```python
    # request period
    if forward_call.src_hotkey in self.request_timestamps:
        period = time.time() - self.request_timestamps[forward_call.src_hotkey][0]
        # Convert the period to hours
        period_hours = period / 3600
        # If the number of requests exceeds the maximum requests per hour
        if len(self.request_timestamps[forward_call.src_hotkey]) > self.config.miner.blacklist.requests_per_hour * period_hours:
            return True, f"{forward_call.src_hotkey} request frequency exceeded {len(self.request_timestamps[forward_call.src_hotkey])} requests in {self.config.miner.blacklist.requests_per_hour} hours."
```

Reply:
This should be equivalent, yes?

```python
    # request period
    if forward_call.src_hotkey in self.request_timestamps:
        period = time.time() - self.request_timestamps[forward_call.src_hotkey][0]
        # Calculating max_request_period for 'x' requests per hour
        max_request_period = 3600 / self.config.miner.blacklist.requests_per_hour
        if period < max_request_period:
            return True, f"{forward_call.src_hotkey} request frequency exceeded {len(self.request_timestamps[forward_call.src_hotkey])} requests in the last hour."
```

However does this mean we wouldn't need the cache and it's param?

Reply:
Let's roll with this for now and open a separate PR to consolidate the params.